idea/src/org/jetbrains/kotlin/idea/inspections/JavaCollectionsStaticMethodCallInspection.kt

```diff
@@ -0,0 +1,123 @@
+import org.jetbrains.kotlin.idea.quickfix.checkUpdateRuntime
+import org.jetbrains.kotlin.load.java.descriptors.JavaMethodDescriptor
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+
+class JavaCollectionsStaticMethodInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : KtVisitorVoid() {
+            override fun visitDotQualifiedExpression(expression: KtDotQualifiedExpression) {
+                super.visitDotQualifiedExpression(expression)
+
+                val callExpression = expression.callExpression ?: return
+                val args = callExpression.valueArguments
+                val firstArg = args.firstOrNull() ?: return
+                val context = expression.analyze(BodyResolveMode.PARTIAL)
+                if (KotlinBuiltIns.FQ_NAMES.mutableList !=
+                        firstArg.getArgumentExpression()?.getType(context)?.constructor?.declarationDescriptor?.fqNameSafe) return
+
+                val resolvedCall = expression.getResolvedCall(context) ?: return
+                val descriptor = resolvedCall.resultingDescriptor as? JavaMethodDescriptor ?: return
+                val fqName = descriptor.importableFqName?.asString() ?: return
+                if (!canReplaceWithStdLib(expression, fqName, args)) return
+
+                val methodName = fqName.split(".").last()
+                holder.registerProblem(expression,
+                                       "Java Collections static method call can be replaced with Kotlin stdlib",
+                                       ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                                       ReplaceWithStdLibFix(methodName, firstArg.text))
+            }
+        }
+    }
+
+    private fun canReplaceWithStdLib(expression: KtDotQualifiedExpression, fqName: String, args: List<KtValueArgument>): Boolean {
+        if (!fqName.startsWith("java.util.Collections.")) return false
+        val size = args.size
+        return when (fqName) {
+            "java.util.Collections.fill" -> checkApiVersion(ApiVersion.KOTLIN_1_2, expression) && size == 2
+            "java.util.Collections.reverse" -> size == 1
+            "java.util.Collections.shuffle" -> checkApiVersion(ApiVersion.KOTLIN_1_2, expression) && (size == 1 || size == 2)
+            "java.util.Collections.sort" -> size == 1 || size == 2
+            else -> false
+        }
+    }
+
+    private fun checkApiVersion(requiredVersion: ApiVersion, expression: KtDotQualifiedExpression): Boolean {
+        val module = ModuleUtilCore.findModuleForPsiElement(expression) ?: return true
+        val version = getRuntimeLibraryVersion(module) ?: return true
```

Comment:
Please use `module.languageVersionSettings.apiVersion` instead of this process (this honors the API version specified in the Gradle build or facet settings).